llvm/test/Transforms/Attributor/pointer-info.ll

```diff
@@ -52,12 +52,34 @@ define void @bar(ptr noundef byval(%struct.test.a) align 8 %dev) {
   store i32 1, ptr %1
   ret void
 }
+
+
+define void @multiple_access_bins_for_select_when_pass_through_instruction(i1 %cond) {
+; CHECK: Function Attrs: mustprogress nofree norecurse nosync nounwind willreturn memory(write)
+; CHECK-LABEL: define {{[^@]+}}@multiple_access_bins_for_select_when_pass_through_instruction
+; CHECK-SAME: (i1 [[COND:%.*]]) #[[ATTR2:[0-9]+]] {
+; CHECK-NEXT:    [[TMP1:%.*]] = alloca [10 x i8], align 4
+; CHECK-NEXT:    [[TMP2:%.*]] = getelementptr inbounds [10 x i8], ptr [[TMP1]], i64 0, i64 5
+; CHECK-NEXT:    [[TMP3:%.*]] = getelementptr inbounds [10 x i8], ptr [[TMP1]], i64 0, i64 2
+; CHECK-NEXT:    [[TMP4:%.*]] = select i1 [[COND]], ptr [[TMP2]], ptr [[TMP3]]
+; CHECK-NEXT:    store i8 100, ptr [[TMP4]], align 1
+; CHECK-NEXT:    ret void
+;
+%2 = alloca [10 x i8], align 4
+%3 = getelementptr inbounds [10 x i8], ptr %2, i64 0, i64 5
+%4 = getelementptr inbounds [10 x i8], ptr %2, i64 0, i64 2
+%5 = select i1 %cond, ptr %3, ptr %4
+store i8 100, ptr %5
+ret void
+}
```

Comment:
I don't get this test. For one, all the code is actually dead. Adding tests like that is never a good idea. Second, it doesn't seem we did compute anything different before, at least the test output looks the same, no?

Reply:
I don't know of a case that will show up differently in IR with just AApointer info. 
With AAalloca info, yes, but that's not implemented in the PR. 
Is there a way to print any debug output that might show the difference?

Reply:
UPDATE: 
1.) Tried an example where a load could be optimized away but that doesn't work due to weak may store logic. 
2.) Debug info doesn't change with or without the change since it's a change to a local internal map that stores the stats of the bins.